src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -755,6 +758,43 @@ namespace winrt::TerminalApp::implementation
         _ApplyTheme(_settings->GlobalSettings().GetTheme());
     }
 
+    fire_and_forget AppLogic::_ApplyStartupTaskStateChange()
+    {
+        co_await winrt::resume_foreground(_root->Dispatcher());

```

Comment:
You'll probably need to get a weak ref to `this` before the `co_await`, and make sure you can get a strong ref afterwards to make sure that `this` is still alive.

We do a lot of that in `TerminalPage` like so:
```c++
        auto weakThis{ get_weak() };
        co_await winrt::resume_foreground(Dispatcher());
        if (auto page{ weakThis.get() })
        {
            // Do something with this
        }
```

Reply:
I'm happy to put that change in. Why is there the need to get the weak _this_ ref? The other `fire_and_forget` routines in the same class don't follow that same pattern -- does it make sense to create an issue to refactor those to grab the weak refs as well?

ex in AppLogic:
```
    fire_and_forget AppLogic::_RefreshThemeRoutine()
    {
        co_await winrt::resume_foreground(_root->Dispatcher());

        // Refresh the UI theme
        _ApplyTheme(_settings->GlobalSettings().GetTheme());
    }
```

Reply:
This _particular_ one we're less worried about. `AppLogic` controls the lifetime of the entire application, so it's actually _never_ going away, and we don't need to really worry about it. Though, you're right that it wouldn't _hurt_ to do the right thing there as well.

Reply:
Created #6295 to track it